addons/web_editor/models/ir_attachment.py

```diff
@@ -60,3 +62,53 @@ def _get_media_info(self):
         """Return a dict with the values that we need on the media dialog."""
         self.ensure_one()
         return self._read_format(['id', 'name', 'description', 'mimetype', 'checksum', 'url', 'type', 'res_id', 'res_model', 'public', 'access_token', 'image_src', 'image_width', 'image_height', 'original_id'])[0]
+
+    def _get_html_fields(self):
+        return [('ir.ui.view', 'arch_db', [('type', '=', 'qweb')])]
+
+    def _is_used_in_html_field(self):
+        """Returns a model where this attachment is used, or a dict with model names or False."""
+        self.ensure_one()
+        # Keep significant part of attachment url
+        url = self.local_url
+        is_default_local_url = url.startswith('/web/image/%s?' % self.id)
+        if is_default_local_url:
+            url = ('/web/image/%s' if self.image_src else '/web/content/%s') % self.id
+
+        # in-document URLs are html-escaped, a straight search will not
+        # find them
+        url = tools.html_escape(url)
+        big_models = []
+        sudo_models = []
+        for model_name, field_name, domain in self._get_html_fields():
+            if '%s,%s' % (model_name, field_name) not in FULL_SCAN_FIELDS:
+                if model_name in big_models:
+                    continue
+                if self.env[model_name].sudo().with_context(active_test=False).search_count([]) > SCAN_MAX_COUNT:
+                    big_models.append(model_name)
+                    continue
+            if is_default_local_url:
+                domain.extend([
+                    "|", "|", "|", "|",
+                    (field_name, 'like', '"%s-' % url),
+                    (field_name, 'like', "'%s-" % url),
+                    (field_name, 'like', '"%s?' % url),
+                    (field_name, 'like', "'%s?" % url),
+                ])
+            domain.extend([
+                "|",
+                (field_name, 'like', '"%s"' % url),
+                (field_name, 'like', "'%s'" % url),
+            ])
+            matches = self.env[model_name].with_context(active_test=False).search(domain)
+            if matches:
+                return matches
+            if model_name in sudo_models:
+                continue
+            sudo_matches = self.env[model_name].sudo().with_context(active_test=False).search(domain)
+            if sudo_matches:
+                sudo_models.append(model_name)
+        return {
+            'skipped_models': big_models,
+            'access_models': sudo_models,
+        } if big_models or sudo_models else False
```

Comment:
We have to consider retro compatibility, that doesn't seem right, this is a change of what the method is returning.
Maybe we need to deprecate this method and use a new one, or simply add a new context param to still use the old code for the caller without that new context key?
We can discuss it tomorrow directly at the office it will be easier and faster.